src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -102,7 +102,25 @@ class WP_Plugin_Dependencies {
 	 * @since 6.5.0
 	 */
 	public static function initialize() {
-		self::read_dependencies_from_plugin_headers();
+		$known_dependencies = get_site_option( 'wp_plugin_dependencies_known_dependencies', array() );
+
+		if ( empty( $known_dependencies ) ) {
+			self::read_dependencies_from_plugin_headers();
+		} else {
+			self::$dependencies = array_filter( $known_dependencies );
+
+			foreach ( self::$dependencies as $plugin => $dependencies ) {
+				self::$dependency_slugs = array_merge(
+					self::$dependency_slugs,
+					$dependencies
+				);
+
+				foreach ( $dependencies as $dependency ) {
+					self::$dependent_slugs[ $dependency ][] = $plugin;
+				}
+			}
```

Comment:
Needs a `self::$dependency_slugs = array_unique( self::$dependency_slugs );` after the loops have finished. Will add this later.

Reply:
Done, and also added `self::$dependent_slugs[ $dependency ] = array_unique( self::$dependent_slugs[ $dependency ] );`

👍 👍